scanpipe/pipes/d2d_config.py

```diff
@@ -0,0 +1,167 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# http://nexb.com and https://github.com/aboutcode-org/scancode.io
+# The ScanCode.io software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode.io is provided as-is without warranties.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# Data Generated with ScanCode.io is provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+# ScanCode.io should be considered or used as legal advice. Consult an Attorney
+# for any legal advice.
+#
+# ScanCode.io is a free software code scanning tool from nexB Inc. and others.
+# Visit https://github.com/aboutcode-org/scancode.io for support and download.
+
+
+from dataclasses import dataclass
+from dataclasses import field
+
+
+@dataclass
+class EcosystemConfig:
+    """
+    Base class for ecosystem-specific configurations to be defined
+    for each ecosystem.
+    """
+
+    # This should be defined for each ecosystem which
+    # are options in the pipelines
+    ecosystem_option: str = "Default"
+
+    # These are extensions for packages of this ecosystem which
```

Comment:
```
    # These are extensions for package archive files for this ecosystem
    # that are matchable against the purldb using matchcode
    matchable_package_extensions: list = field(default_factory=list)

    # These are extensions for file of this ecosystem that are
    # matchable against the purldb using matchcode
    matchable_resource_extensions: list = field(default_factory=list)
```
... this may be a better name?